pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -900,15 +947,59 @@ func TestDesiredRouterDeploymentSingleReplica(t *testing.T) {
 		t.Errorf("expected replicas to be 1, got %d", *deployment.Spec.Replicas)
 	}
 
-	if deployment.Spec.Template.Spec.Affinity != nil {
-		t.Errorf("expected no affinity, got %+v", *deployment.Spec.Template.Spec.Affinity)
-	}
-
```

Comment:
Did you mean to remove this?  I don't see this change in #1089.

Reply:
Iirc this was another issue with nil pointer in the backport.

Reply:
Does it just need an `if` guard?  Better to add an `if` guard than to drop the check entirely.  If you need to drop the check entirely for whatever reason, please note so in the PR description and commit message so that it doesn't look like an accident.

Reply:
Fixed.  You're right, it was an oversight.